### DISCUSSION TITLE
Clickhouse on Kubernetes: is that recommended or not?

### DISCUSSION THREAD

**Discussion by collimarco** [ClickHouse/ClickHouse#38557] [General]
Hello,

Does it make sense / would you recommend to run Clickhouse on Kubernetes?
Or Clickhouse is not really made for that and it is better to run it on VM / bare servers?
In particular I wonder if Clickhouse performance optimizations (which are the real value of Clickhouse) would work well in a containerized environment.

Thanks

**Comment by nikitamikhaylov** [comment]
Containerised environment is a very thin layer between ClickHouse and host OS, so all optimisations will work as expected. Everything will be Ok.
